### PR TITLE
chore: deprecate func compile_uri_template()

### PR DESCRIPTION
**Pull request by nix010**
# Summary of Changes

First timer, I don't know if this is a correct for this so please advise. 

Add deprecated warning for func compile_uri_template()

# Related Issues
#1967

# Pull Request Checklist

This is just a reminder about the most common mistakes.  Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [x] Applied changes to both WSGI and ASGI code paths and interfaces (where applicable).
- [x] Added **tests** for changed code.
- [x] Prefixed code comments with GitHub nick and an appropriate prefix.
- [x] Coding style is consistent with the rest of the framework.
- [x] Updated **documentation** for changed code.
    - [x] Added docstrings for any new classes, functions, or modules.
    - [x] Updated docstrings for any modifications to existing code.
    - [x] Updated both WSGI and ASGI docs (where applicable).
    - [x] Added references to new classes, functions, or modules to the relevant RST file under `docs/`.
    - [x] Updated all relevant supporting documentation files under `docs/`.
    - [x] A copyright notice is included at the top of any new modules (using your own name or the name of your organization).
    - [x] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/usage/restructuredtext/directives.html?highlight=versionadded#directive-versionadded).
- [x] Changes (and possible deprecations) have [towncrier](https://towncrier.readthedocs.io/en/actual-freaking-docs/index.html) news fragments under `docs/_newsfragments/`, with the file name format `{issue_number}.{fragment_type}.rst`. (Run `towncrier --draft` to ensure it renders correctly.)

If you have *any* questions to *any* of the points above, just **submit and ask**! This checklist is here to *help* you, not to deter you from contributing!

*PR template inspired by the attrs project.*

### CODE REVIEW

**Comment by CaselIT** [comment]
Thanks for the pr!

**Comment by CaselIT** [comment]
> Furthermore, we need to either wrap all tests in `tests/test_uri_templates_legacy.py` with `pytest.warns(...)`, or add a rule to ignore this warning, because we don't want to get spam in the test output about this.

it may make sense to make pytest fail the test if a warning gets raised, so that's explicitly handled

**Comment by vytas7** [comment]
@nix010 Just checking whether you are still working on this PR? We'd need to clean up the tests etc to make all CI gates pass as a first step, and check there are no new warnings introduced.

File: falcon/routing/util.py [hunk 1]
```diff
@@ -27,8 +28,9 @@ def __init__(self, message):
 
 
 # NOTE(kgriffs): Published method; take care to avoid breaking changes.
+@deprecated("")
 def compile_uri_template(template):
-    """Compile the given URI template string into a pattern matcher.
+    """*Deprecated soon* Compile the given URI template string into a pattern matcher.
```
[the remark this answers is not in the document]
Comment (vytas7) [reply falcon/routing/util.py]: For now, we could just follow what we already have.
Which is adding a `Note`, or just describing this function as deprecated. See, e.g., https://falcon.readthedocs.io/en/stable/api/request_and_response_wsgi.html#falcon.Response.body

**Review by vytas7 — APPROVED** [review]
Thanks once again for this PR! (I've taken the liberty of tweaking this somewhat.)

**Review by CaselIT — APPROVED** [review]

**Pull request merged**